### DISCUSSION TITLE
Run on Different PORt

### DISCUSSION THREAD

**Discussion by Aniket524** [horilla-opensource/horilla#437] [General]
Can We Run the Application on the different ports?

**Comment by horilla-opensource** [comment]
Here’s a response you can use for the GitHub discussion:  

---

Hi @Aniket524 ,

Yes, you can run the **Horilla HRMS** application on different ports. Here's how you can do it:  

### Using Django's Built-in Development Server  
1. **Specify the Port**:  
   When starting the development server, you can specify the desired port.  
   ```bash
   python manage.py runserver 0.0.0.0:<desired_port>
   ```  
   For example, to run it on port 8080:  
   ```bash
   python manage.py runserver 0.0.0.0:8080
   ```  

2. **Default Port**:  
   If no port is specified, Django runs on port **8000** by default.  

### Using a Production Web Server  
If you're running the application in production using **Gunicorn**, **uWSGI**, or another WSGI server, you can configure the port in the respective server's configuration.  

For example, with **Gunicorn**:  
```bash
gunicorn horilla.wsgi:application --bind 0.0.0.0:<desired_port>
```  

### Considerations  
- Ensure the port you choose is not already in use by another application.  
- Update your firewall settings if necessary to allow traffic on the new port.  
- If using a reverse proxy (e.g., **NGINX** or **Apache**), make sure it’s configured to forward traffic to the correct port.  

Let us know if you need further guidance on setting up the application!  

With Regards,
Team Horilla
